src/table/utils/handle-click.ts

```diff
@@ -28,3 +30,40 @@ export const handleMouseDownLabelToFocusHeadCell = (evt: MouseEvent, rootElement
   evt.preventDefault();
   updateFocus({ focusType: 'focus', cell: getCellElement(rootElement, [0, columnIndex]) });
 };
+
+export const getMouseHandlers = (
+  cell: Cell,
+  announce: Announce,
+  onClick: React.MouseEventHandler<HTMLTableCellElement> | undefined,
+  selectionDispatch: SelectionDispatch,
+  isFlagEnabled: (flag: string) => boolean
+) => {
+  const selectMultiEnabled = isFlagEnabled('PS_15585_SN_TABLE_BASIC_FEATURES');
+
+  const handleMouseDown = (evt: React.MouseEvent) => {
+    // Prevent focus to be set to the cell, done on mouse up instead
+    evt.preventDefault();
+    evt.stopPropagation();
+    if (selectMultiEnabled && cell.isSelectable)
+      selectionDispatch({ type: SelectionActions.SELECT_MULTI_START, payload: { cell } });
+  };
+
+  const handleMouseOver = (evt: React.MouseEvent) => {
+    if (selectMultiEnabled && evt.button === 0 && evt.buttons === 1)
+      selectionDispatch({ type: SelectionActions.SELECT_MULTI_ADD, payload: { cell, evt, announce } });
+  };
+
+  const handleMouseUp = (evt: React.MouseEvent) => {
+    // update focusCellCoord and manually focus the element
+    onClick?.(evt as React.MouseEvent<HTMLTableCellElement>);
+    (evt.target as HTMLElement).focus();
```

Comment:
What's up with the forced casting?

Reply:
I think we need to re-visit the typing for the events. I think the target here is not always a proper element, unless you specify that. In this case we know what the target is and can assume that `focus` is on there